### ISSUE TITLE
Not getting lamda exceptions for social login

### ISSUE THREAD

**Issue by Sunsiha** [aws-amplify/amplify-flutter#1279]
When I tried to call social login we are doing few lamda functions. But in flutter am not able to catch the lamba exceptions for social login. When I tried to login with social auth providers getting error like this;

`social loginAuthException(message: Sign-in with web UI failed, recoverySuggestion: See attached exception for more details, underlyingException: com.amazonaws.mobileconnectors.cognitoauth.exceptions.AuthServiceException: invalid_request)`

Code:
```
  try {
    var res = await Amplify.Auth.signInWithWebUI(
        provider: authProvider,
        options: const CognitoSignInWithWebUIOptions(
          isPreferPrivateSession: true,
        ));
    print("return response$res");
  } on AuthException catch (e) {
  }
```

Note: For other cases am getting lamda exceptions  in terminal only for social login am not getting. But this lamda errors are getting in web. And we are using same configurations.

They are handling like this;
```
else if (error.toString().includes("EMAIl_ALREADY_EXISTS")) {
    setErr("Email already exists. Please try again with new one!");
}
```

**Expected Result:**
Should be able to get the lamda exceptions.

**Flutter details:**
```
Flutter 2.5.3 • channel stable • https://github.com/flutter/flutter.git
Framework • revision 18116933e7 (3 months ago) • 2021-10-15 10:46:35 -0700
Engine • revision d3ea636dc5
Tools • Dart 2.14.4
```
**Cognito versions:**
```
amplify_flutter: ^0.2.10
amplify_auth_cognito: ^0.2.10
```
Added the triggers also;

<img width="814" alt="Screenshot 2022-01-19 at 1 02 57 AM" src="https://user-images.githubusercontent.com/27992382/150006241-f373be44-b192-4eba-b2bb-ae9769c6bdb0.png">
<img width="1153" alt="Screenshot 2022-01-19 at 1 03 24 AM" src="https://user-images.githubusercontent.com/27992382/150006314-cba96add-eb1c-4c3c-bd3a-e226defadad5.png">

**Comment by dnys1** [comment]
Hi @Sunsiha - I will try to reproduce. It's possible that this is not supported but I will confirm.

**Comment by Sunsiha** [comment]
> Hi @Sunsiha - I will try to reproduce. It's possible that this is not supported but I will confirm.

You mean this exception itself we won't get?

**Comment by dnys1** [comment]
Hi @Sunsiha - I was able to reproduce the issue, and only when using social sign in (not with hosted UI or regular sign in/sign up). Would you be willing to create an issue in the `amplify-android` repo with all the information here? They may have more context as to why this is.

**Comment by Sunsiha** [comment]
> Hi @Sunsiha - I was able to reproduce the issue, and only when using social sign in (not with hosted UI or regular sign in/sign up). Would you be willing to create an issue in the `amplify-android` repo with all the information here? They may have more context as to why this is.

I need a solution. But the same issue is available in iOS also. So again I have to create new issue in amplify-ios also?

**Comment by dnys1** [comment]
I'm very sorry - as we depend on these libraries internally, I'm not able to resolve the problem without their assistance. I can create the tickets, if you'd prefer.

**Comment by Sunsiha** [comment]
> I'm very sorry - as we depend on these libraries internally, I'm not able to resolve the problem without their assistance. I can create the tickets, if you'd prefer, although for their sake it may be easier if you did since you will be able to troubleshoot and respond to their feedback.

Please do create a ticket and let me know

**Comment by dnys1** [comment]
I've created the issue tickets. Please follow them so that you receive the latest updates.

- https://github.com/aws-amplify/amplify-android/issues/1649
- https://github.com/aws-amplify/amplify-ios/issues/1617

**Comment by Sunsiha** [comment]
> I've created the issue tickets. Please follow them so that you receive the latest updates.
> 
> * [[Auth] Lambda trigger exception not forwarded when using social sign-in amplify-android#1649](https://github.com/aws-amplify/amplify-android/issues/1649)
> * [[Auth] Lambda trigger exception not forwarded when using social sign-in amplify-ios#1617](https://github.com/aws-amplify/amplify-ios/issues/1617)

Okay thanks.

**Comment by Sunsiha** [comment]
any update?

**Comment by Jordan-Nelson** [comment]
Hello @Sunsiha - We have recently released a new developer preview version of the Auth category ([amplify_flutter v1.0.0-next.0](https://pub.dev/packages/amplify_flutter/versions/1.0.0-next.0)). This new version is a full re-write of the auth category in dart, and it supports Mobile, Desktop, and Web. This issue is addressed in this new version.

You can read more about the developer preview release in the [blog post](https://aws.amazon.com/blogs/mobile/amplify-flutter-auth-web-and-desktop/) and [docs](https://docs.amplify.aws/lib/auth/getting-started/q/platform/flutter/). I will update this issue again once this new version is out of developer preview.

**Comment by Jordan-Nelson** [comment]
Hi @Sunsiha - Pub did not recognize the first dev preview we released as supporting web. We have released a revision which is recognized by Pub as supporting all platforms. You can see that [here](https://pub.dev/packages/amplify_flutter/versions/1.0.0-next.0+1).

The original version supported all platforms as well, Pub just didn't recognize as supporting all platforms because it wasn't listed in the pubspec. We have updated that in the latest release.

**Comment by AsitDixit** [comment]
> Hi @Sunsiha - Pub did not recognize the first dev preview we released as supporting web. We have released a revision which is recognized by Pub as supporting all platforms. You can see that [here](https://pub.dev/packages/amplify_flutter/versions/1.0.0-next.0+1).
> 
> The original version supported all platforms as well, Pub just didn't recognize as supporting all platforms because it wasn't listed in the pubspec. We have updated that in the latest release.

When this fix will release for amplify_android . for fluter it was released on feb 24 2022 but no response for android. I'm facing this issue on Android. I have created an issue [https://github.com/aws-amplify/amplify-android/issues/1907](url)

**Comment by Jordan-Nelson** [comment]
This issue has been addressed in v1.0.0 of Amplify Flutter, which is now stable. This release also includes web and desktop support for Auth, API, Analytics, and Storage. You can see the list of new features and bug fixes in the [release notes](https://github.com/aws-amplify/amplify-flutter/releases/tag/v1.0.0), and see more details on how to migrate in the [upgrade guide](https://docs.amplify.aws/lib/project-setup/upgrade-guide/q/platform/flutter/).
